server/permalinks.go

```diff
@@ -98,6 +98,20 @@ func (p *Plugin) processReplacement(r replacement, glClient *gitlab.Client, wg *
 		Ref: &r.permalinkData.commit,
 	}
 	projectPath := fmt.Sprintf("%s/%s", r.permalinkData.user, r.permalinkData.repo)
+
+	// Check if project is public
+	if p.getConfiguration().EnableCodePreview == "public" {
```

Comment:
do we need constants for this ?

Reply:
I don't think so, as we are not using it anywhere else